packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -442,6 +442,28 @@ class FuchsiaDevice extends Device {
     }
   }
 
+  @override
+  bool get supportsScreenshot => true;
+
+  @override
+  Future<void> takeScreenshot(File outputFile) async {
+    if (outputFile.basename.split('.').last != 'ppm') {
+      throw '${outputFile.path} must be a .ppm file';
+    }
+    final RunResult screencapResult = await shell('screencap > /tmp/screenshot.ppm');
+    if (screencapResult.exitCode != 0) {
+      throw 'Could not take a screenshot on device $name';
```

Comment:
I think it will probably be nice to have some more info here, like:
`throw 'Could not take a screenshot on device $name':\n$screencapResult';`
or
`throw 'Could not take a screenshot on device $name':\n${screencapResult.stderr}';`